src/gtest/test_sidechain_to_mempool.cpp

```diff
@@ -241,58 +242,58 @@ TEST_F(SidechainsInMempoolTestSuite, hasSidechainCreationTxTest) {
     uint256 scId = uint256S("1492");
 
     //Case 1: no sidechain related tx in mempool
-    bool res = aMempool.hasSidechainCreationTx(scId);
+    bool res = aMempool->hasSidechainCreationTx(scId);
     EXPECT_FALSE(res);
 
     //Case 2: fwd transfer tx only in mempool
     CTransaction fwdTx = GenerateFwdTransferTx(scId, CAmount(10));
     CTxMemPoolEntry fwdPoolEntry(fwdTx, /*fee*/CAmount(1), /*time*/ 1000, /*priority*/1.0, /*height*/1987);
-    aMempool.addUnchecked(fwdPoolEntry.GetTx().GetHash(), fwdPoolEntry);
-    res = aMempool.hasSidechainCreationTx(scId);
+    aMempool->addUnchecked(fwdPoolEntry.GetTx().GetHash(), fwdPoolEntry);
+    res = aMempool->hasSidechainCreationTx(scId);
     EXPECT_FALSE(res);
 
     //Case 3: btr tx only in mempool
     CTransaction btrTx = GenerateBtrTx(scId);
     CTxMemPoolEntry btrTxEntry(btrTx, /*fee*/CAmount(1), /*time*/ 1000, /*priority*/1.0, /*height*/1987);
-    aMempool.addUnchecked(btrTxEntry.GetTx().GetHash(), btrTxEntry, /*fCurrentEstimate*/true);
-    res = aMempool.hasSidechainCreationTx(scId);
+    aMempool->addUnchecked(btrTxEntry.GetTx().GetHash(), btrTxEntry, /*fCurrentEstimate*/true);
+    res = aMempool->hasSidechainCreationTx(scId);
     EXPECT_FALSE(res);
 
     //Case 4: sc creation tx in mempool
     CTransaction scTx  = GenerateScTx(CAmount(10));
     const uint256& scIdOk = scTx.GetScIdFromScCcOut(0);
     CTxMemPoolEntry scPoolEntry(scTx, /*fee*/CAmount(1), /*time*/ 1000, /*priority*/1.0, /*height*/1987);
-    aMempool.addUnchecked(scPoolEntry.GetTx().GetHash(), scPoolEntry);
-    res = aMempool.hasSidechainCreationTx(scIdOk);
+    aMempool->addUnchecked(scPoolEntry.GetTx().GetHash(), scPoolEntry);
+    res = aMempool->hasSidechainCreationTx(scIdOk);
     EXPECT_TRUE(res);
 }
 
 TEST_F(SidechainsInMempoolTestSuite, ScAndFwdsAndBtrInMempool_ScNonRecursiveRemoval) {
-    // Associated scenario: Sidechain creation and some fwds and btr are in mempool.
```

Comment:
I guess this is one too many replacements here! 😄

Reply:
Oh, right, a minor flaw in the otherwise brilliant `replace_all` idea :)